Src/Library/Binder/RequestBinder.cs

```diff
@@ -155,29 +155,28 @@ private static void BindRouteValues(TRequest req, RouteValueDictionary routeValu
     private static void BindQueryParams(TRequest req, IQueryCollection query, List<ValidationFailure> failures)
     {
         if (query.Count == 0) return;
-        //var cachedProps = fromQueryProps;
-        //foreach (var prop in cachedProps)
-        //{
-        //    var dictionary = new Dictionary<string, object>();
-        //    foreach (var nestedProp in prop.PropType.GetProperties())
-        //    {
-        //        if (query.TryGetValue(nestedProp.Name.ToLower(), out var values))
-        //        {
-        //            var parser = nestedProp.PropertyType.ValueParser();
-        //            if (parser != null)
-        //            {
-        //                var (success, nestedValue) = parser(values);
-        //                if (success)
-        //                {
-        //                    dictionary.Add(nestedProp.Name, nestedValue);
-        //                }
-        //            }
-        //        }
-        //    }
-        //    var json = JsonSerializer.Serialize(dictionary);
-        //    var value = JsonSerializer.Deserialize(json, prop.PropType)!;
-        //    prop.PropSetter(req, value);
-        //}
+        var cachedProps = fromQueryProps;
+        foreach (var prop in cachedProps)
+        {
+            var value = prop.ConstructorInfo!.Invoke(Array.Empty<object>());
```

Comment:
this is definitely a no no. constructorinfo invoke is very slow. that is why we're using compiled and cached expressions. it's gonna be a nightmare to try and do compiled/cached expressions for all nested property types.

Reply:
Ok. I'll try to reuse query model with request inheritance then. Thank you for your time!